omod/src/main/java/org/openmrs/module/metrics/builder/Impl/JmxReportBuilderImpl.java

```diff
@@ -0,0 +1,44 @@
+package org.openmrs.module.metrics.builder.Impl;
+
+import javax.management.MBeanServer;
+import java.lang.management.ManagementFactory;
+import java.time.LocalDateTime;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.jmx.JmxReporter;
+import com.codahale.metrics.jvm.JvmAttributeGaugeSet;
+import com.codahale.metrics.jvm.MemoryUsageGaugeSet;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.module.metrics.api.service.MetricService;
+import org.openmrs.module.metrics.builder.JmxReportBuilder;
+import org.openmrs.module.metrics.model.MetricscConfigImpl;
+import org.springframework.beans.factory.annotation.Autowired;
+
+public class JmxReportBuilderImpl implements JmxReportBuilder {
+	
+	protected final Log log = LogFactory.getLog(this.getClass());
+	
+	@Autowired
+	private MetricService metricService;
+	
+	@Override
+	public MetricRegistry initializeMetricRegistry() {
```

Comment:
Instead of this relatively static configuration of metrics, wouldn't it make sense to define the `MetricRegsitry` as a Spring bean allowing this to be configured through either Java code (by injecting the `MetricRegistry` and calling `register`) or XML?

Reply:
This looks ok, but we are going to need to add some custom gaugesets to make this useful.